### PR TITLE
Extend Baseline agent-installer steps for the BM labs automation - Add Static network configuration

### PR DESCRIPTION
**Pull request by bmanzari**
Add support for static network configurations in agent based installations on bare metal

### CODE REVIEW

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-compact-ipv4-static-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    dns-resolver:
+          config:
+            server:
+              - ${INTERNAL_NET_IP}
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: ${INTERNAL_NET_IP}
+          next-hop-interface: ${baremetal_iface}
+          table-id: 254
```
[the remark this answers is not in the document]
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh:75]: removed

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    interfaces:
+    - name: ${baremetal_iface}
+      type: ethernet
+      state: up
+      ipv4:
+        enabled: ${ipv4_enabled}
+        dhcp: false
+        address:
+            - ip: ${ip}
+              prefix-length: 23
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh:41]: this must be set correctly. 

use ` ${INTERNAL_NET_CIDR##*/}`

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      documentation: "Whether to enable the ipv4 stack"
+    - name: ipv6_enabled
+      default: "true"
+      documentation: "Whether to enable the ipv6 stack"
+    - name: AUX_HOST
+      default: ""
+      documentation: ""
+    - name: INTERNAL_NET_IP
+      default: "192.168.90.1"
+      documentation: ""
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-ref.yaml:25]: Add

```
  - name: INTERNAL_NET_CIDR
    default: "192.168.90.0/24"
```

Ref: https://github.com/openshift/release/pull/37113/files#diff-1a25efbd40c4c56b7204ce724811447fc67366b6dac42a6f2ab15a1588b17dc1R41

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh (hunk 2 above)
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh:41]: like this " prefix-length: ${INTERNAL_NET_CIDR##*/} " ?

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        enabled: false
+        dhcp: false
+      ipv6:
+        enabled: false
+        dhcp: false
+    "
+  done
+  
+  # Take care of the indentation when adding the routes to the above yaml
+  # TODO check what happens with multiple baremetal_ifaces, we are not looping here
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh:64]: You'd remove this todo, interfaces are under loops as needed. baremetal_iface is expected to be the one interface that provides outbound networks reachability. An "additional networks" handling can come later if there are scenarios requiring it, but that would be another feature.

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh (hunk 2 above)
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh]: changed

File: ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh (hunk 4 above)
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/conf/static/baremetal-lab-agent-conf-static-commands.sh]: removed

**Comment by bmanzari** [comment]
Compact: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37113/rehearse-37113-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-compact-ipv4-static-connected-p1-f14/1634124620778967040

SNO: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37113/rehearse-37113-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-connected-p1-f14/1633924262970527744

**Review by aleskandro — APPROVED** [review]

**Pull request merged**